xapian-letor/bin/xapian-prepare-trainingfile.cc

```diff
@@ -0,0 +1,119 @@
+/** @file xapian-prepare-trainingfile.cc
+ * @brief Command line tool to prepare Letor training file
+ */
+/* Copyright (C) 2004,2005,2006,2007,2008,2009,2010,2015 Olly Betts
+ * Copyright (C) 2011 Parth Gupta
+ * Copyright (C) 2016 Ayush Tomar
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301
+ * USA
+ */
+
+#include <config.h>
+
+#include <xapian.h>
+#include <xapian-letor.h>
+
+#include <iostream>
+#include <string>
+
+#include "gnu_getopt.h"
+
+using namespace std;
+
+#define PROG_NAME "xapian-prepare-trainingfile"
+#define PROG_DESC "Command line tool to prepare letor training file"
+
+#define OPT_HELP 1
+#define OPT_VERSION 2
+
+static void show_usage() {
+    cout << "Usage: " PROG_NAME " [OPTIONS] <queryfile> <qrelfile> <trainingfile>\n"
+    "Options:\n"
+    "  -d, --db=DIRECTORY  database to search (multiple databases may be specified)\n"
+    "  -m, --msize=MSIZE   maximum number of matches to return\n"
+    "      --help          display this help and exit\n"
+    "      --version       output version information and exit\n";
+}
+
+int
+main(int argc, char **argv)
+try {
```

Comment:
I think you may be mixing up two things.

There doesn't need to be an extra block around the `try` here, but the body of the `try` should still be indented.

There was a case I mentioned earlier where we typically don't indent, but that was for the lines which are just strings in the help text in `show_usage()` above (in other programs we don't indent them really just because they're formatted to fit in 80 columns, and without an indent the text looks much more like it will once output - in particular if your editor window is 80 columns wide, you can see where the lines are too long easily) - e.g. look at `xapian-core/bin/xapian-delve.cc`.